### DISCUSSION TITLE
The VPN service 'org.freedesktop.NetworkManager.strongswan' was not installed

### DISCUSSION THREAD

**Discussion by suntong** [strongswan/strongswan#2444] [Q&A] [answered]
Hi, 

I'm trying to setup IKEv2 VPN clients under Kubuntu, and followed with [How to setup IKEv2 VPN on Linux](https://www.securevpn.pro/eng/setup/linux-ikev2-vpn), but I wan't able to make the VPN connection in the end, only getting a pop up saying

`The VPN service 'org.freedesktop.NetworkManager.strongswan' was not installed.`

The required configuration is really _really_ simple, the following is _all_ it takes for me to get my IKEv2 VPN working under Windows:

![image](https://github.com/user-attachments/assets/409ec76e-7e66-45e8-9f73-3947d28e8325)

However, I was having a hard time getting the same thing working under Linux -- The problem is that there's no any info on how to check/verify the configuration on the config file and cli level (the above article is just a series of screenshots with no verification/troubleshooting info at all), so I have no clue how to debug. Trying to solve it myself, I found an article, [Configure Linux VPN clients using the command line](https://github.com/hwdsl2/setup-ipsec-vpn/blob/master/docs/clients.md#configure-linux-vpn-clients-using-the-command-line), but it seems to be using an entirely different set of tools than the above GUI version, which is the route I want to take.

Trying to answer question from https://github.com/strongswan/strongswan/discussions/196, and all other misc checking that I found on the net:

```sh
$ plocate ipsec.conf | wc 
      0       0       0

$ plocate swanctl.conf | wc 
      0       0       0

$ plocate strongswan | head -3
.../archives/libstrongswan_5.9.13-2ubuntu4_amd64.deb
.../archives/network-manager-strongswan_1.6.0-3build2_amd64.deb
.../archives/strongswan-libcharon_5.9.13-2ubuntu4_amd64.deb

$ plocate strongswan.conf
/usr/share/man/man5/strongswan.conf.5.gz
/usr/share/strongswan/templates/config/strongswan.conf
/var/lib/dpkg/info/libstrongswan.conffiles

$ cat /usr/lib/NetworkManager/conf.d/10-globally-managed-devices.conf
[keyfile]
unmanaged-devices=*,except:type:wifi,except:type:gsm,except:type:cdma

$ ipsec statusall
-bash: ipsec: command not found

$ sudo ipsec statusall
sudo: ipsec: command not found

$ swanctl -L
-bash: swanctl: command not found

$ sudo swanctl -L
sudo: swanctl: command not found
```

Checking `/var/log/syslog` yields no more clues except for only this line:

`NetworkManager[726]: <info>  [1725554533.9653] audit: op="connection-activate" uuid="3a86edea-...d4643c" name="my vpn" pid=1765 uid=xxyy result="fail" reason="The VPN service 'org.freedesktop.NetworkManager.strongswan' was not installed."`

And I've rebooted my machine many may times.   
Any help appreciated.

Notes:

- This is about **Kubuntu**, so any other solution like using `network-manager-l2tp-gnome` etc will not be helpful.
- The [How to setup IKEv2 VPN on Linux](https://www.securevpn.pro/eng/setup/linux-ikev2-vpn) is want I want, as 
it uses username & password, which is what I need; whereas [this article](https://github.com/hwdsl2/setup-ipsec-vpn/blob/master/docs/ikev2-howto.md#linux) requires _"Before configuring Linux VPN clients, you must make the following change on the VPN server"_, which is impossible for my case. 
- I've installed all packages mentioned in above two articles, here is the list:

```
ii libcharon-extauth-plugins  amd64 strongSwan charon library (extended authentication plugins)
ii libstrongswan  amd64 strongSwan utility and crypto library
ii network-manager-strongswan  amd64 network management framework (strongSwan plugin)
ii strongswan-libcharon  amd64 strongSwan charon library
ii strongswan-nm  amd64 strongSwan plugin to interact with NetworkManager
```

- The last one, `strongswan-nm` was only installed when the first attempt failed, but it seems not helping either. 

```sh
$ apt-cache policy network-manager-strongswan
network-manager-strongswan:
  Installed: 1.6.0-3build2
  Candidate: 1.6.0-3build2
  Version table:
 *** 1.6.0-3build2 500
        500 http://archive.ubuntu.com/ubuntu noble/universe amd64 Packages
        100 /var/lib/dpkg/status

$ apt-cache policy libstrongswan
libstrongswan:
  Installed: 5.9.13-2ubuntu4
  Candidate: 5.9.13-2ubuntu4
  Version table:
 *** 5.9.13-2ubuntu4 500
        500 http://archive.ubuntu.com/ubuntu noble/main amd64 Packages
        100 /var/lib/dpkg/status

$ uname -rm 
6.8.0-41-generic x86_64

$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 24.04 LTS
Release:        24.04
Codename:       noble
```

**Comment by suntong** [comment]
> I'm trying to setup IKEv2 VPN clients under Kubuntu,

This seems to be a common request that I found to have no good answer, like 

- ["I am using the standard network-manager in Kubuntu 14.04, I am unsure what additional packages I installed "](https://www.kubuntuforums.net/forum/archives/eol-releases/-14-04/network-support-ar/62277-l2tp-ipsec-vpn-connection)
- ["How to set up VPN using IKEv2?"](https://askubuntu.com/questions/1433565/how-to-set-up-vpn-using-ikev2)
- ["The VPN service 'org.freedesktop.NetworkManager.ssh' was not installed"](https://www.kubuntuforums.net/forum/archives/eol-releases/-18-04/network-support-ax/70922-the-vpn-service-org-freedesktop-networkmanager-ssh-was-not-installed)
- ["can't set up a vpnc connection with network-manager on Kubuntu 15.04"](https://askubuntu.com/questions/646441/cant-set-up-a-vpnc-connection-with-network-manager-on-kubuntu-15-04)
- ["VPN settings not working -- I had to dump Linux due to not being able to reliably use a pptp VPN, I tried everything under the sun. I have also seen an interesting thread on some other computing forum where a professional Linux web server admin made it very clear how he never used Linux at his home, only Windows due to VPN connection setups being so unreliable in Linux but so 100% reliable in Win 7."](https://www.kubuntuforums.net/forum/archives/eol-releases/-13-10/network-support-aq/58006-vpn-settings-not-working)

**Comment by tobiasbrunner** [comment]
The article your refer to uses the GNOME plugin for NetworkManager, which is [maintained by us](https://docs.strongswan.org/docs/5.9/features/networkManager.html and what you install via `network-manager-strongswan` package. So you won't need this.

However, for KDE, there is a separate plugin for the Plasma version of NetworkManager, which is not maintained by us. As far as I can tell, this GUI is provided directly with the `plasma-nm` package.

Either way, you need the `strongswan-nm` package as that provides the actual D-Bus service that NetworkManager interacts with (charon-nm), whether it's configured via GNOME or KDE. Why that wouldn't work (i.e. why that service is not found), I don't know. Maybe some issue with KDE/Kubuntu.

**Reply by suntong** [reply]
Thanks for the quick reply. Let me clarify -- My understanding from you above paragraph is that  I need to, 

- uninstall `network-manager-strongswan` while keep/reinstall `strongswan-nm`
- install `charon-nm`?? right?

**UPDATE:** the `strongswan-nm` plugin "provides an interface which allows NetworkManager to configure
 and control the IKEv2 daemon directly through D-Bus. It is designed to work
 in conjunction with the network-manager-strongswan package, providing
 a simple graphical frontend to configure IPsec based VPNs." so I should not uninstall `network-manager-strongswan`, right?

That second step might be what I was missing...

And, any idea how to check/verify the configuration on the config file and cli level, or how to enable more debug logging pls?

thx

**Reply by tobiasbrunner** [reply, marked as answer]
> so I should not uninstall `network-manager-strongswan`, right?

No, you should. It's completely useless unless you use a GTK-based NetworkManager frontend. You need `strongswan-nm`, though, as that provides `charon-nm`.

> And, any idea how to check/verify the configuration on the config file and cli level, or how to enable more debug logging pls?

No config files or CLI tools of strongSwan are involved if you use the NetworkManager plugin. You can see the logs of `charon-nm` via `journalctl -u NetworkManager` (it logs via syslog by default, so if you want to change the log levels, use `charon-nm.syslog.daemon.*`, see [the docs](https://docs.strongswan.org/docs/5.9/config/logging.html) for details).
